### PR TITLE
Migrate from parse_url() to wp_parse_url()

### LINKED ISSUE #6801: Migrate from `parse_url` to `wp_parse_url()`
https://github.com/WordPress/WordPress/blob/9f4bbcdb7896a7baba9eb88add281f3fbcdec0ef/wp-includes/http.php#L628-L682

### PR DESCRIPTION
**Pull request by shantanu2704**
Fixes #6801

#### Changes proposed in this Pull Request:

Replace all occurrences of parse_init with wp_parse_init

### CODE REVIEW

**Comment by shantanu2704** [comment]
Thanks, Derek!

File: class.json-api-endpoints.php [hunk 1]
```diff
@@ -1842,7 +1842,7 @@ function handle_media_sideload( $url, $parent_post_id = 0, $type = 'any' ) {
 			return false;
 
 		// if we didn't get a URL, let's bail
-		$parsed = @parse_url( $url );
+		$parsed = @wp_parse_url( $url );
```
Comment (zinigor) [inline class.json-api-endpoints.php]: Let's remove output silencing here because it's already done inside `wp_parse_url`.

File: class.photon.php [hunk 2]
```diff
@@ -840,12 +840,12 @@ public function filter_sizes( $sizes, $size ) {
 	 * @return bool
 	 */
 	protected static function validate_image_url( $url ) {
-		$parsed_url = @parse_url( $url );
+		$parsed_url = @wp_parse_url( $url );
```
Comment (zinigor) [inline class.photon.php]: Same here, let's remove silencing.

File: functions.photon.php [hunk 3]
```diff
@@ -208,7 +208,7 @@ function jetpack_photon_url( $image_url, $args = array(), $scheme = null ) {
 add_filter( 'jetpack_photon_pre_args', 'jetpack_photon_parse_wpcom_query_args', 10, 2 );
 
 function jetpack_photon_parse_wpcom_query_args( $args, $image_url ) {
-	$parsed_url = @parse_url( $image_url );
+	$parsed_url = @wp_parse_url( $image_url );
```
Comment (zinigor) [inline functions.photon.php]: Same here, let's remove silencing.

File: json-endpoints/class.wpcom-json-api-edit-media-v1-2-endpoint.php [hunk 4]
```diff
@@ -146,7 +146,7 @@ private function build_file_array_from_url( $media_id, $url ) {
 		}
 
 		// if we didn't get a URL, let's bail
-		$parsed = @parse_url( $url );
+		$parsed = @wp_parse_url( $url );
```
Comment (zinigor) [inline json-endpoints/class.wpcom-json-api-edit-media-v1-2-endpoint.php]: Please remove silencing.

File: modules/protect/shared-functions.php [hunk 5]
```diff
@@ -211,8 +211,8 @@ function jetpack_clean_ip( $ip ) {
 		$ip = $matches[1];
 	}
 
-	if ( function_exists( 'parse_url' ) ) {
-		$parsed_url = parse_url( $ip );
+	if ( function_exists( 'wp_parse_url' ) ) {
+		$parsed_url = wp_parse_url( $ip );
```
Comment (enejb) [inline modules/protect/shared-functions.php]: When would this not be there?

**Pull request closed without merging**